controllers/argocd/deployment.go

```diff
@@ -1102,3 +1130,15 @@ func isDexDisabled() bool {
 	}
 	return false
 }
+
+// to update nodeSelector and tolerations in reconciler
+func updateNodePlacement(existing *appsv1.Deployment, deploy *appsv1.Deployment, changed *bool) {
+	if !reflect.DeepEqual(existing.Spec.Template.Spec.NodeSelector, deploy.Spec.Template.Spec.NodeSelector) {
+		existing.Spec.Template.Spec.NodeSelector = deploy.Spec.Template.Spec.NodeSelector
```

Comment:
I remember we want to setup Node Affinity rather than Node selector as per @shubhamagarwal19 suggestion. So that we can implement `requiredDuringSchedulingIgnoredDuringExecution` and `preferredDuringSchedulingIgnoredDuringExecution`. Is there any change in the plan ?

https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity

Reply:
I did go through the node-affinity concept, it is different from nodeSelector, and the issue https://github.com/argoproj-labs/argocd-operator/issues/342 specifically asks for nodeSelector and Tolerations